product-scenarios/3-securing-an-api-with-different-security-protocols/3.1-api-developers-want-to-secure-their-api-using-an-authentication-machanism/3.1.1-api-developers-and-publishers-want-to-make-their-apis-secure-using-the-oauth2-protocol/src/test/java/org/wso2/am/scenario/tests/api/secure/oauth2/SecureUsingAuth2TestCases.java

```diff
@@ -20,58 +20,142 @@
+        verifyResponse(addSubscriptionResponse);
+        log.info(TEST_APPLICATION_NAME + " is subscribed to " + TEST_API_1_NAME);
+
+        accessToken = generateAppKeys();
     }
 
+    @Test(description = "3.1.1.1", dataProvider = "AuthorizationHeadersDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testOAuth2Authorization(String tokenPrefix) throws Exception {
+        Map<String, String> requestHeaders = new HashMap<>();
+        requestHeaders.put(APIMIntegrationConstants.AUTHORIZATION_HEADER, tokenPrefix + accessToken);
+        requestHeaders.put("Content-Type", "application/x-www-form-urlencoded");
+        String gatewayHttpsUrl = getHttpsAPIInvocationURL(TEST_API_1_CONTEXT, TEST_API_1_VERSION, "/CheckPhoneNumber");
+        log.debug("Gateway HTTPS URL : " + gatewayHttpsURL);
+        List<NameValuePair> urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair("PhoneNumber", "18006785432"));
+        urlParameters.add(new BasicNameValuePair("LicenseKey", "0"));
+        HttpResponse apiResponse = HttpClient.doPost(gatewayHttpsUrl, requestHeaders, urlParameters);
+        log.debug("API response : " + apiResponse.getData());
+        assertEquals(apiResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Response code mismatched when api invocation");
+    }
 
-    @AfterClass(alwaysRun = true)
-    public void destroy() throws Exception {
-        for (String name : applicationsList) {
-            apiStore.removeApplication(name);
+    public void createApplication(String applicationName) throws APIManagerIntegrationTestException {
+        HttpResponse addApplicationResponse = null;
+        try {
+            addApplicationResponse = apiStore.addApplication(applicationName,
+                    APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, "", "description");
+        } catch (APIManagerIntegrationTestException e) {
+            String error = "Error in creating Testing Application " + applicationName;
+            throw new APIManagerIntegrationTestException(error, e);
         }
-        applicationsList.clear();
-        deleteUser(SUBSCRIBER_USERNAME, ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
-        deleteUser(API_DEVELOPER_USERNAME,  ADMIN_LOGIN_USERNAME, ADMIN_PASSWORD);
-        apiPublisher.deleteAPI("PizzaShackTestAPI", "1.0.0", API_DEVELOPER_USERNAME);
+        verifyResponse(addApplicationResponse);
+        log.info("Application - " + applicationName + "is created successfully");
     }
 
-    @Test(description = "1.1.1.1")
-    public void testOAuth2Authorization() throws Exception {
-        // create and publish sample API
-        apiPublisher.developSampleAPI("swaggerFiles/pizzashack-swagger.json", API_DEVELOPER_USERNAME, backendEndPoint,
-                true, "public");
-        //TO DO
+    public String generateAppKeys() throws APIManagerIntegrationTestException {
```

Comment:
Let's not throw APIManagerIntegrationTestException instead we can just throw exception

Reply:
Fixed with 750e7bd10c57712fcaa878a3f17915c8da88fee7

Reply:
Fixed with 1ec043717269d7bca1c17f860256c55d0a973aa0